### ISSUE TITLE
Unexpected result:  '6..{$2=\noubomn}' matches field with no $2

### ISSUE THREAD

**Issue by danmichaelo** [MARCspec/File_MARC_Reference#9]
I'm getting unexpected matches from the marcspec `6..{$2=\noubomn}`. To reproduce, here's my setup:

```bash
$ composer init -q --require="ck/file_marc_reference:~1.1" --require="pear/file_marc:~1.1" && composer install
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 4 installs, 0 updates, 0 removals
  - Installing pear/pear_exception (v1.0.0): Loading from cache
  - Installing pear/file_marc (1.1.5): Loading from cache
  - Installing ck/php-marcspec (v1.7.2): Loading from cache
  - Installing ck/file_marc_reference (v1.1.1): Loading from cache
pear/file_marc suggests installing pear/validate_ispn (Install optionally via your project's composer.json)
Writing lock file
Generating autoload files
```

and a small test script `test.php`:

```php
<?php
# test.php
require('vendor/autoload.php');

$data = '<?xml version="1.0" encoding="UTF-8"?>
<record xmlns="">
  <datafield tag="650" ind1=" " ind2="7">
    <subfield code="a">Partielle differensialligninger</subfield>
    <subfield code="2">noubomn</subfield>
  </datafield>
  <datafield tag="650" ind1=" " ind2="7">
    <subfield code="a">Numeriske metoder</subfield>
    <subfield code="2">noubomn</subfield>
  </datafield>
  <datafield tag="650" ind1=" " ind2="7">
    <subfield code="a">Industriell matematikk (Kongresser)</subfield>
    <subfield code="2">tekord</subfield>
  </datafield>
  <datafield tag="653" ind1=" " ind2=" ">
    <subfield code="a">numeriske</subfield>
    <subfield code="a">metoder</subfield>
    <subfield code="a">elementmetoder</subfield>
    <subfield code="a">partielle</subfield>
    <subfield code="a">differensialligninger</subfield>
    <subfield code="a">industriell</subfield>
    <subfield code="a">matematikk</subfield>
  </datafield>
</record>';

$spec = '6..{$2=\noubomn}';

$rec = (new File_MARCXML($data, File_MARCXML::SOURCE_STRING))->next();

$results = new File_MARC_Reference($spec, $rec);
foreach ($results->data as $res) {
	echo $res . "\n\n";
}
```

Running this, I get:

```bash
$ php test.php
650  7 _aPartielle differensialligninger
       _2noubomn

650  7 _aNumeriske metoder
       _2noubomn

653    _anumeriske
       _ametoder
       _aelementmetoder
       _apartielle
       _adifferensialligninger
       _aindustriell
       _amatematikk
```

My question is: why did `6..{$2=\noubomn}` match the 653 field, which doesn't have a `$2` subfield? Is this a bug, or a problem with my marcspec understanding?

**Comment by cKlee** [comment]
Must be a bug. I will investigate.

**Comment by cKlee** [comment]
This should be solved in v1.1.4. Is it working for you?

**Comment by danmichaelo** [comment]
Seems to work perfect, thanks!
